p2p/memp2p/memp2p.go

```diff
@@ -351,7 +351,14 @@ func (messenger *Messenger) ReceiveMessage(topic string, message p2p.MessageP2P)
 		messenger.Network.LogMessage(message)
 	}
 
-	err := validator.ProcessReceivedMessage(message)
+	var handler func(buffToSend []byte)
+	if allowBroadcast {
+		handler = func(buffToSend []byte) {
+			messenger.Broadcast(topic, buffToSend)
+		}
+	}
+
+	err := validator.ProcessReceivedMessage(message, handler)
 
 	return err
```

Comment:
return directly. no need to save in error

Reply:
changed